src/sentry/api_gateway/proxy.py

```diff
@@ -0,0 +1,51 @@
+"""
+Utilities related to proxying a request to a region silo
+"""
+
+
+from django.http import StreamingHttpResponse
+from requests import Response as ExternalResponse
+from requests import request as external_request
+from rest_framework.exceptions import NotFound
+from rest_framework.request import Request
+
+from sentry.models.organization import Organization
+
+# stream 0.5 MB at a time
+PROXY_CHUNK_SIZE = 512 * 1024
+
+
+def _parse_response(response: ExternalResponse) -> StreamingHttpResponse:
+    """
+    Convert the Responses class from requests into the drf Response
+    """
+    streamed_response = StreamingHttpResponse(
+        streaming_content=(chunk for chunk in response.iter_content(PROXY_CHUNK_SIZE)),
```

Comment:
I think putting this behind a generator would be what you want to do to take advantage of the stream. 

Example:

https://github.com/getsentry/sentry/blob/c02e62db05f68cdffbeb3c7279d9b2d227dbcbe4/src/sentry/profiles/utils.py#L93-L97

Reply:
@dashed 
Incidentally, that syntax is an iterator comprehension and is largely identical to your generator form.  The syntax is subtle but a comprehension that lives in `[]` is a list comprehension whilst a `()` is a list comprehension.

The import part wrt streaming is the `pull`, non buffered aspects of this, which I believe do work as written.

Reply:
One other detail that we should keep in mind here however.  Timeouts!
@AniketDas-Tekky Do we know what the default timeouts inside our requests client look like?  Here are some timeouts we need to keep in mind:
timeout for connect from client
timeout for body read from server (say a client holds on to the body for 10 seconds without completing the read, does the server break the connection?)
timeout for body read from client (say a client holds on to the connection but hasn't received the entire content-length worth of data and the server does not close the connection.   how long does the client keep the connection alive?)

These are going to matter in various network failure modes.  We should be very mindful moving forward to select defaults that make sense.

Reply:
@corps so right now I don't think there is any timeout for requests, but I think we'll want to configure one eventually.

Reply:
Absolutely.  it may also even make sense to setup sentry_options that allow us to dynamically configure this through OPS.  When we get to the point of having, say, EU regions that we're gating through, *we absolutely will encounter all kinds of networking partition issues* both temporal and worse in some cases, so getting this right matters (we don't want cascading failures by clogging up control servers pending on proxy requests).

In fact, one additional thing I ponder is how concurrency is going to work for these proxied endpoints through django.  Not a django expert, but will django allow us to use python's Futures with say, green threads or events, so we're not stuck holding onto a pending proxied connection via a live thread?  Ideally, the server doing the proxying is using the absolute minimum overhead to maintain this opaque pass through operations.

Reply:
> The syntax is **_subtle_** but a comprehension that lives in `[]` is a list comprehension whilst a `()` is an iterator comprehension.

@corps this is why I've learned to prefer clarity over cleverness regardless of language familiarity. :D It helps with the long-term maintenance of code bases as teams change over time.

Reply:
Fair enough, I have no stake in the game other than to clarify the code as it was.

Reply:
Changed it